src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,161 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        linkTail(value);
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == 0) {
+            linkHead(value);
+        } else if (index == size) {
+            linkTail(value);
+        } else {
+            Node<T> currentNode = findNode(index);
+            Node<T> newNode = new Node<>(currentNode.prev, value, currentNode);
+            currentNode.prev.next = newNode;
+            currentNode.prev = newNode;
+            size++;
+        }
+
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            linkTail(value);
```

Comment:
not critical, but it makes more sense
```suggestion
            add(value);
```